### PR TITLE
protoc.Wrapper

### PR DESCRIPTION
**Pull request by izumin5210**
To improve testability...

### CODE REVIEW

File: pkg/grapicmd/di/providers.go [hunk 1]
```diff
@@ -70,23 +77,37 @@ func ProvideGexConfig(ctx *grapicmd.Ctx) *gex.Config {
...
 }
 
-func ProvideInitializeProjectUsecase(ctx *grapicmd.Ctx, gexCfg *gex.Config, ui clui.UI, generator module.Generator) usecase.InitializeProjectUsecase {
-	return usecase.NewInitializeProjectUsecase(
-		ui,
-		generator,
-		gexCfg,
-		ctx.Version,
-	)
+func ProvideToolRepository(ctx *grapicmd.Ctx, gexCfg *gex.Config) (tool.Repository, error) {
```
Comment (watchbunny) [inline pkg/grapicmd/di/providers.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideToolRepository should have comment or be unexported

File: pkg/grapicmd/di/providers.go [hunk 2]
```diff
@@ -70,23 +77,37 @@ func ProvideGexConfig(ctx *grapicmd.Ctx) *gex.Config {
...
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+	}
+	return toolRepo, nil
 }
 
-func ProvideExecuteProtocUsecase(ctx *grapicmd.Ctx, gexCfg *gex.Config, ui clui.UI, executor command.Executor, generator module.Generator) usecase.ExecuteProtocUsecase {
-	return usecase.NewExecuteProtocUsecase(
+func ProvideProtocWrapper(ctx *grapicmd.Ctx, ui clui.UI, toolRepo tool.Repository) protoc.Wrapper {
```
Comment (watchbunny) [inline pkg/grapicmd/di/providers.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideProtocWrapper should have comment or be unexported

File: pkg/grapicmd/di/providers.go [hunk 3]
```diff
@@ -70,23 +77,37 @@ func ProvideGexConfig(ctx *grapicmd.Ctx) *gex.Config {
...
 		ui,
-		executor,
-		gexCfg,
+		toolRepo,
 		ctx.RootDir,
+		ctx.BinDir,
+	)
+}
+
+func ProvideInitializeProjectUsecase(ctx *grapicmd.Ctx, gexCfg *gex.Config, ui clui.UI, generator module.Generator) usecase.InitializeProjectUsecase {
```
Comment (watchbunny) [inline pkg/grapicmd/di/providers.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function ProvideInitializeProjectUsecase should have comment or be unexported

File: pkg/protoc/wrapper.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	"github.com/izumin5210/gex/pkg/tool"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"k8s.io/utils/exec"
+
+	"github.com/izumin5210/grapi/pkg/clui"
+	"github.com/izumin5210/grapi/pkg/grapicmd/util/fs"
+)
+
+type Wrapper interface {
```
Comment (watchbunny) [inline pkg/protoc/wrapper.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type Wrapper should have comment or be unexported

File: pkg/protoc/wrapper.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	cfg      *Config
+	fs       afero.Fs
+	ui       clui.UI
+	execer   exec.Interface
+	toolRepo tool.Repository
+	rootDir  string
+	binDir   string
+}
+
+func NewWrapper(cfg *Config, fs afero.Fs, execer exec.Interface, ui clui.UI, toolRepo tool.Repository, rootDir, binDir string) Wrapper {
```
Comment (watchbunny) [inline pkg/protoc/wrapper.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function NewWrapper should have comment or be unexported

File: pkg/protoc/wrapper.go [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+func (e *wrapperImpl) execProtocAll(ctx context.Context) error {
+	protoFiles, err := e.cfg.ProtoFiles(e.fs, e.rootDir)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	var errs []error
+	for _, path := range protoFiles {
+		err = e.execProtoc(ctx, path)
+		relPath, _ := filepath.Rel(e.rootDir, path)
```
Comment (watchbunny) [inline pkg/protoc/wrapper.go]: **[errcheck]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
	relPath, _ := filepath.Rel(e.rootDir, path)

Comment (izumin5210) [reply pkg/protoc/wrapper.go]: 👌

File: pkg/protoc/wrapper.go (hunk 5 above)
Comment (izumin5210) [reply pkg/protoc/wrapper.go]: fixed in d26e797

File: pkg/protoc/wrapper.go (hunk 4 above)
Comment (izumin5210) [reply pkg/protoc/wrapper.go]: fixed in d26e797

**Pull request merged**